kairos-data/src/transaction.rs

```diff
@@ -0,0 +1,137 @@
+use chrono::{Utc, NaiveDateTime};
+use diesel::prelude::*;
+use crate::schema::transactions;
+use serde::{Deserialize, Serialize};
+use hex;
+
+use kairos_circuit_logic::transactions::{KairosTransaction, Signed, Transfer, Withdraw, L1Deposit};
+
+const TRANSFER_TRX: i16 = 1;
+const DEPOSIT_TRX: i16 = 2;
+const WITHDRAW_TRX: i16 = 3;
+
+#[derive(Queryable, Debug, Identifiable, Insertable, Serialize, Selectable)]
+#[diesel(primary_key(timestamp, amount, public_key))]
+#[diesel(table_name = transactions)]
+#[diesel(check_for_backend(diesel::pg::Pg))]
+pub struct Transaction {
+    pub timestamp: NaiveDateTime,
+    pub public_key: String,
+    pub nonce: Option<i64>,
+    pub trx: i16,
+    pub amount: i64,
+    pub recipient: Option<String>,
+}
```

Comment:
Can we normalize this table.
That would remove the need for the enum tag `trx`.

Reply:
We could, but it adds needless complication querying with little benefit. Nothing needs to be done from the API side for the crate to figure out which enum it should be.

Reply:
So `trx` is either 0,1,2 mapped to (deposit, transfer, withdrawal)? lgtm.